iap_local_receipt/iap_pkcs7_verifier.py

```diff
@@ -1,104 +1,86 @@
-from M2Crypto import SMIME, X509, BIO, Err, m2
-from M2Crypto.SMIME import PKCS7
+from OpenSSL import crypto
+from cffi import FFI
+
+ffi = FFI()
 
 
 class PKCS7VerifyError(Exception):
-    pass
+    def __init__(self, message=None, **kwargs):
+        super(PKCS7VerifyError, self).__init__(
+            message or ffi.string(crypto._lib.ERR_error_string(crypto._lib.ERR_get_error(), ffi.NULL)),
+            **kwargs
+        )
 
 
 class PKCS7Verifier:
     def __init__(self, root_ca_cert_file=None, root_ca_cert_string=None):
-        self.smime = SMIME.SMIME()
-
+        self.store = None
         if root_ca_cert_file:
             self.load_ca_cert_file(root_ca_cert_file)
         elif root_ca_cert_string:
             self.load_ca_cert_string(root_ca_cert_string)
 
-        # OpenSSL will find the signer's cert in the PKCS7 data,
-        # but SMIME.verify still needs a stack of signer certs,
-        # even if it is empty, which is what we do here.
-        self.smime.set_x509_stack(X509.X509_Stack())
-
     def load_ca_cert_file(self, ca_cert_file):
         """
         Load a CA cert from a PEM file, replacing any previous cert.
         """
-        cert = X509.load_cert(ca_cert_file)
-        self._load_cert(cert)
+        self.load_ca_cert_string(open(ca_cert_file, 'rb').read())
 
     def load_ca_cert_string(self, ca_cert_string):
         """
         Load a CA cert from a PEM string, replacing any previous cert.
         """
-        cert = X509.load_cert_string(ca_cert_string)
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, ca_cert_string)
         self._load_cert(cert)
 
     def _load_cert(self, cert):
-        st = X509.X509_Store()
-        st.add_cert(cert)
-        self.smime.set_x509_store(st)
+        self.store = crypto.X509Store()
+        self.store.add_cert(cert)
 
-    def verify_file(self, pkcs7_der_file):
+    def verify_file(self, pkcs7_der_file, verify_time=None):
         """
         Verify signature on signed PKCS7 DER file.
         Return blob containing the signed data.
         Throw PKCS7VerifyError if verification failed.
         This will fail if the CA cert has not been loaded.
         """
-        # Load the data, verify it.
-        p7 = load_pkcs7_der(pkcs7_der_file)
-        return self._verify(p7)
+        return self.verify_data(open(pkcs7_der_file, 'rb').read(), verify_time)
 
-    # This will fail if the ca cert has not been loaded
-    def verify_data(self, pkcs7_der):
+    def verify_data(self, pkcs7_der, verify_time=None):
         """
         Verify signature on signed PKCS7 DER blob.
         Return blob containing the signed data.
         Throw PKCS7VerifyError if verification failed.
         This will fail if the CA cert has not been loaded.
         """
+        store = self.store or crypto.X509Store()
+        if verify_time:
+            store.set_time(verify_time)
```

Comment:
The function didn't have docstrings for `pkcs7_der` either so I didn't assume you guys were in the habit of adding them.